### PR TITLE
versioning metadata service

### PR DESCRIPTION
**Pull request by ad-47**
Proto change: https://github.com/VertaAI/modeldb/pull/118

### CODE REVIEW

**Comment by conradoverta** [comment]
@ravishetye @evgeniilezhnin reminder to review

File: backend/src/main/java/ai/verta/modeldb/entities/metadata/LabelsMappingEntity.java [hunk 1]
```diff
@@ -13,57 +16,67 @@
 public class LabelsMappingEntity {
   public LabelsMappingEntity() {}
 
-  public LabelsMappingEntity(String entityHash, Integer entityType, String label) {
-    this.id = new LabelMappingId(entityHash, entityType, label);
+  public LabelsMappingEntity(IdentificationType id, String label) {
+    if (id.getIdCase().getNumber() == 2) {
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/entities/metadata/LabelsMappingEntity.java]: can we use enum Vlaues instead on indices?

File: backend/src/main/java/ai/verta/modeldb/entities/metadata/LabelsMappingEntity.java [hunk 2]
```diff
@@ -13,57 +16,67 @@
...
+  public static class LabelMappingId implements Serializable {
 
-  @Column(name = "entity_type", length = 50)
-  private Integer entity_type;
+    @Column(name = "label", length = 50)
+    private String label;
 
-  @Column(name = "entity_hash", nullable = false, columnDefinition = "varchar", length = 32)
-  private String entity_hash;
+    @Column(name = "entity_type", length = 50)
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/entities/metadata/LabelsMappingEntity.java]: is length attribute used? is so what does 50 represent?

File: backend/src/main/java/ai/verta/modeldb/entities/metadata/LabelsMappingEntity.java [hunk 3]
```diff
@@ -13,57 +16,67 @@
...
+    private Integer entity_type;
 
-  public LabelMappingId(String entityHash, Integer entityType, String label) {
-    this.entity_hash = entityHash;
-    this.entity_type = entityType;
-    this.label = label;
-  }
+    @Column(name = "entity_hash", nullable = false, columnDefinition = "varchar", length = 32)
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/entities/metadata/LabelsMappingEntity.java]: @evgeniilezhnin 's proposal to change length will affect this? Please coordinate amongst yourself  and fix

Comment (ad-47) [reply backend/src/main/java/ai/verta/modeldb/entities/metadata/LabelsMappingEntity.java]: Changed by @evgeniilezhnin on other PR: https://github.com/VertaAI/modeldb/pull/132/files#diff-d59c40fe7da8470d549f26f156a36415R32

File: backend/src/main/java/ai/verta/modeldb/entities/metadata/LabelsMappingEntity.java (hunk 2 above)
Comment (ad-47) [reply backend/src/main/java/ai/verta/modeldb/entities/metadata/LabelsMappingEntity.java]: Removed

File: backend/src/main/java/ai/verta/modeldb/entities/metadata/LabelsMappingEntity.java (hunk 1 above)
Comment (ad-47) [reply backend/src/main/java/ai/verta/modeldb/entities/metadata/LabelsMappingEntity.java]: Changed

**Pull request merged**